### PR TITLE
🌟 Add support for Django2

### PR DESCRIPTION
**Pull request by dodobas**

### CODE REVIEW

File: temba/api/v1/tests.py [hunk 1]
```diff
@@ -1364,7 +1364,9 @@ def test_api_contacts(self):
 
         # try to post a new group with a blank name
         response = self.postJSON(url, dict(phone="+250788123456", groups=["  "]))
-        self.assertResponseError(response, "groups", "This field may not be blank.")
+        body = response.json()
```
Comment (dodobas) [inline temba/api/v1/tests.py]: DRF 3.8.x has improved error handling for nested fields: https://github.com/encode/django-rest-framework/pull/5655/files

File: temba/api/v2/tests.py [hunk 2]
```diff
@@ -174,65 +174,65 @@ def test_serializer_fields(self):
         self.assertRaises(serializers.ValidationError, field.to_internal_value, list(range(101)))  # too long
 
         field = fields.CampaignField(source="test")
-        field.context = {"org": self.org}
+        field._context = {"org": self.org}
```
Comment (dodobas) [inline temba/api/v2/tests.py]: Field context is a readonly property so we need to set it directly as *protected* attribute

File: temba/locations/models.py [hunk 3]
```diff
@@ -12,13 +12,13 @@
 
 
 # default manager for AdminBoundary, doesn't load geometries
-class NoGeometryManager(models.GeoManager):
+class NoGeometryManager(models.Manager):
```
Comment (dodobas) [inline temba/locations/models.py]: Geomanager was removed ... https://docs.djangoproject.com/en/2.0/releases/1.9/#django-contrib-gis

File: temba/middleware.py [hunk 4]
```diff
@@ -5,24 +5,25 @@
...
 from django.utils import timezone, translation
+from django.utils.deprecation import MiddlewareMixin
 
 from temba.contacts.models import Contact
 from temba.orgs.models import Org
 from temba.policies.models import Policy
 
 
-class ExceptionMiddleware(object):
+class ExceptionMiddleware(MiddlewareMixin):
```
Comment (dodobas) [inline temba/middleware.py]: Update to new-style middleware functions

File: temba/settings.py.dev [hunk 5]
```diff
@@ -60,9 +60,9 @@ if DEBUG_TOOLBAR:
 # -----------------------------------------------------------------------------------
 # In development, add in extra logging for exceptions and the debug toolbar
 # -----------------------------------------------------------------------------------
-MIDDLEWARE_CLASSES = ("temba.middleware.ExceptionMiddleware",) + MIDDLEWARE_CLASSES
+MIDDLEWARE = ("temba.middleware.ExceptionMiddleware",) + MIDDLEWARE
```
Comment (dodobas) [inline temba/settings.py.dev]: This was *deprecated* in Django 1.10, removed in Django 2.0 ... https://docs.djangoproject.com/en/2.0/releases/1.10/#new-style-middleware

File: temba/settings_common.py [hunk 6]
```diff
@@ -905,9 +905,11 @@
         "v2.api": "2500/hour",
     },
     "PAGE_SIZE": 250,
+    "DEFAULT_PAGINATION_CLASS": "rest_framework.pagination.LimitOffsetPagination",
```
Comment (dodobas) [inline temba/settings_common.py]: Updated DRF settings are to behave the same as older DRF versions

File: templates/500.html [hunk 7]
```diff
@@ -0,0 +1,72 @@
+<html>
```
Comment (dodobas) [inline templates/500.html]: This was taken from TextIt, and added to fix TemplateNotFound Exception, not sure if we have some other generic 500.html template for RapidPro

Comment (norkans7) [reply templates/500.html]: I think this is fine

File: temba/api/support.py [hunk 8]
```diff
@@ -134,6 +133,8 @@ def temba_exception_handler(exc, context):
     """
     Custom exception handler which prevents responding to API requests that error with an HTML error page
     """
+    from rest_framework.views import exception_handler
```
Comment (norkans7) [inline temba/api/support.py]: why is this moved?

Comment (dodobas) [reply temba/api/support.py]: in new version of DRF this is causing a module import loop internally in the DRF, there is an issue on their repo, i could look for it if anyone is interested

File: temba/api/v2/serializers.py [hunk 9]
```diff
@@ -736,6 +748,9 @@ class FlowRunReadSerializer(ReadSerializer):
     path = serializers.SerializerMethodField()
     values = serializers.SerializerMethodField()
     exit_type = serializers.SerializerMethodField()
+    created_on = serializers.DateTimeField(default_timezone=pytz.UTC)
```
Comment (dodobas) [inline temba/api/v2/serializers.py]: DRF requires that we specify default timezone when serializing data, if not specified it will use timezone defined in Django.settings

File: temba/orgs/views.py [hunk 10]
```diff
@@ -1751,7 +1751,7 @@ def pre_process(self, request, *args, **kwargs):  # pragma: needs cover
                 )
                 return HttpResponseRedirect(reverse("public.public_index"))
 
-            if not request.user.is_authenticated():
+            if not request.user.is_authenticated:
```
Comment (dodobas) [inline temba/orgs/views.py]: `is_authenticated` and `is_anonymous` are now properties on the User model

File: temba/channels/types/external/views.py [hunk 11]
```diff
@@ -56,7 +56,7 @@ class ClaimForm(ClaimViewMixin.Form):
         )
 
         encoding = forms.ChoiceField(
-            Channel.ENCODING_CHOICES,
+            choices=Channel.ENCODING_CHOICES,
```
Comment (dodobas) [inline temba/channels/types/external/views.py]: `choices` are now a mandatory keyword argument
